c/src/core/encoder.c

```diff
@@ -78,7 +78,7 @@ static uint8_t pn_type2code(pn_encoder_t *encoder, pn_type_t type)
   case PN_MAP: return PNE_MAP32;
   case PN_DESCRIBED: return PNE_DESCRIPTOR;
   default:
-    return pn_error_format(pni_encoder_error(encoder), PN_ERR, "not a value type: %u\n", type);
+    return pn_error_format(pni_encoder_error(encoder), PN_ERR, "not a value type: %u\n", (unsigned int)type);
```

Comment:
This is about printing an enum, the compiler is unhappy with `%u`. Not really sure what's the nicest way to resolve this. In C23, it will be possible to specify a backing type for enum.

Nowadays, if I read correctly, iti is either a char, int, or unsigned int. Wouldn't maybe treating it as an int work better here? So %d and (int) cast?

Reply:
Let;s go with %d as well